provider/pdns.go

```diff
@@ -57,6 +62,68 @@ const (
 	retryAfterTime = 250 * time.Millisecond
 )
 
+// PDNSConfig is comprised of the fields necessary to create a new PDNSProvider
+type PDNSConfig struct {
+	DomainFilter DomainFilter
+	DryRun       bool
+	Server       string
+	APIKey       string
+	TLSConfig    TLSConfig
+}
+
+// TLSConfig is comprised of the TLS-related fields necessary to create a new PDNSProvider
+type TLSConfig struct {
+	TLSEnabled            bool
+	CAFilePath            string
+	ClientCertFilePath    string
+	ClientCertKeyFilePath string
+}
+
+func (tlsConfig *TLSConfig) setHTTPClient(pdnsClientConfig *pgo.Configuration) error {
+	if !tlsConfig.TLSEnabled {
+		if tlsConfig.CAFilePath != "" {
+			return errors.New("certificate authority file path was specified, but TLS was not enabled")
+		}
+		if tlsConfig.ClientCertFilePath != "" {
+			return errors.New("client certificate file path was specified, but TLS was not enabled")
+		}
+		if tlsConfig.ClientCertKeyFilePath != "" {
+			return errors.New("client certificate key file path was specified, but TLS was not enabled")
+		}
+		return nil
+	}
+
+	log.Debug("Configuring TLS for PDNS Provider.")
+	if tlsConfig.CAFilePath == "" {
```

Comment:
Here I would check not only CAFilePath. ClientCertFilePath and ClientCertKeyFilePath should be checked as well. Could also be a one line if you don't want to have a specific error for the missing setting:
``
if tlsConfig.CAFilePath == "" ||  tlsConfig.ClientCertFilePath == "" || ...
``
But I'm also fine with the other approach

Reply:
I omitted the checks for the other two, since they're not required, and `tlsutils.NewTLSConfig` includes checks that they are either both or neither specified.

However, I don't mind adding back the check I had in my original commit to be more legible/defensive ([link](https://github.com/kubernetes-incubator/external-dns/pull/607/commits/15b279a4bc2e37402d887c325cae186a8759f219#diff-2359ef28e4197bae56cef42409a4cfd2R97)).  Let me know if you prefer this - going to leave as is, for now.